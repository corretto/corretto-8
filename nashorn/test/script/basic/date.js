/*
 * Copyright (c) 2010, 2013, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

/**
 * Basic checks for Date constructor.
 * FIXME: add more checks
 *
 * @test
 * @option -timezone=Asia/Calcutta
 * @run
 */

// check arity of tricky functions
print("Date.length = " + Date.length);
print("Date.UTC.length = " + Date.UTC.length);
print("Date.prototype.setSeconds.length = " + Date.prototype.setSeconds.length);
print("Date.prototype.setUTCSeconds.length = " + Date.prototype.setUTCSeconds.length);
print("Date.prototype.setMinutes.length = " + Date.prototype.setMinutes.length);
print("Date.prototype.setUTCMinutes.length = " + Date.prototype.setUTCMinutes.length);
print("Date.prototype.setHours.length = " + Date.prototype.setHours.length);
print("Date.prototype.setUTCHours.length = " + Date.prototype.setUTCHours.length);
print("Date.prototype.setMonth.length = " + Date.prototype.setMonth.length);
print("Date.prototype.setUTCMonth.length = " + Date.prototype.setUTCMonth.length);
print("Date.prototype.setFullYear.length = " + Date.prototype.setFullYear.length);
print("Date.prototype.setUTCFullYear.length = " + Date.prototype.setUTCFullYear.length);

function printDate(d) {
    print(d.getMinutes());
    print(d.getSeconds());
    print(d.getMilliseconds());
    print(d.getUTCDate());
    print(d.getUTCDay());
    print(d.getUTCMonth());
    print(d.getUTCFullYear());
    print(d.getUTCHours());
    print(d.getUTCMinutes());
    print(d.getUTCSeconds());
    print(d.getUTCMilliseconds());
    print(d.toISOString());
    print(d.toUTCString());
    print(d.toString());
    print(d.toLocaleString());
    print(d.toLocaleDateString());
    print(d.toLocaleTimeString());
    print(d.toDateString());
    print(d.toTimeString());
    print(d.toJSON());
}

var d = new Date(2011, 4, 3, 17, 1, 1, 0);
printDate(d);

d.setUTCMinutes(40, 34);
printDate(d);

d = new Date(Date.UTC(2000, 10, 1, 1, 1, 1, 1));
printDate(d);

d = new Date(0);
d.setFullYear(2012);
d.setMonth(1);
d.setDate(2);
d.setHours(3);
d.setMinutes(3);
d.setSeconds(4);
d.setMilliseconds(5);
printDate(d);

d = new Date(0);
d.setUTCFullYear(2012);
d.setUTCMonth(1);
d.setUTCDate(2);
d.setUTCHours(3);
d.setUTCMinutes(4);
d.setUTCSeconds(5);
d.setUTCMilliseconds(6);
printDate(d);

d = new Date(0);
d.setTime(1000);
printDate(d);
