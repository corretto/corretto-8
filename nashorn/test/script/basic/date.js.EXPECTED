Date.length = 7
Date.UTC.length = 7
Date.prototype.setSeconds.length = 2
Date.prototype.setUTCSeconds.length = 2
Date.prototype.setMinutes.length = 3
Date.prototype.setUTCMinutes.length = 3
Date.prototype.setHours.length = 4
Date.prototype.setUTCHours.length = 4
Date.prototype.setMonth.length = 2
Date.prototype.setUTCMonth.length = 2
Date.prototype.setFullYear.length = 3
Date.prototype.setUTCFullYear.length = 3
1
1
0
3
2
4
2011
11
31
1
0
2011-05-03T11:31:01.000Z
Tue, 03 May 2011 11:31:01 GMT
Tue May 03 2011 17:01:01 GMT+0530 (IST)
Tue May 03 2011 17:01:01 GMT+0530 (IST)
2011-05-03
17:01:01
Tue May 03 2011
17:01:01 GMT+0530 (IST)
2011-05-03T11:31:01.000Z
10
34
0
3
2
4
2011
11
40
34
0
2011-05-03T11:40:34.000Z
Tue, 03 May 2011 11:40:34 GMT
Tue May 03 2011 17:10:34 GMT+0530 (IST)
Tue May 03 2011 17:10:34 GMT+0530 (IST)
2011-05-03
17:10:34
Tue May 03 2011
17:10:34 GMT+0530 (IST)
2011-05-03T11:40:34.000Z
31
1
1
1
3
10
2000
1
1
1
1
2000-11-01T01:01:01.001Z
Wed, 01 Nov 2000 01:01:01 GMT
Wed Nov 01 2000 06:31:01 GMT+0530 (IST)
Wed Nov 01 2000 06:31:01 GMT+0530 (IST)
2000-11-01
06:31:01
Wed Nov 01 2000
06:31:01 GMT+0530 (IST)
2000-11-01T01:01:01.001Z
3
4
5
1
3
1
2012
21
33
4
5
2012-02-01T21:33:04.005Z
Wed, 01 Feb 2012 21:33:04 GMT
Thu Feb 02 2012 03:03:04 GMT+0530 (IST)
Thu Feb 02 2012 03:03:04 GMT+0530 (IST)
2012-02-02
03:03:04
Thu Feb 02 2012
03:03:04 GMT+0530 (IST)
2012-02-01T21:33:04.005Z
34
5
6
2
4
1
2012
3
4
5
6
2012-02-02T03:04:05.006Z
Thu, 02 Feb 2012 03:04:05 GMT
Thu Feb 02 2012 08:34:05 GMT+0530 (IST)
Thu Feb 02 2012 08:34:05 GMT+0530 (IST)
2012-02-02
08:34:05
Thu Feb 02 2012
08:34:05 GMT+0530 (IST)
2012-02-02T03:04:05.006Z
30
1
0
1
4
0
1970
0
0
1
0
1970-01-01T00:00:01.000Z
Thu, 01 Jan 1970 00:00:01 GMT
Thu Jan 01 1970 05:30:01 GMT+0530 (IST)
Thu Jan 01 1970 05:30:01 GMT+0530 (IST)
1970-01-01
05:30:01
Thu Jan 01 1970
05:30:01 GMT+0530 (IST)
1970-01-01T00:00:01.000Z
